localstack/plugin/manager.py

```diff
@@ -0,0 +1,316 @@
+import logging
+import threading
+from typing import Any, Callable, Dict, Generic, List, Tuple, TypeVar, Union
+
+from .core import Plugin, PluginFinder, PluginLifecycleListener, PluginSpec, PluginSpecResolver
+
+LOG = logging.getLogger(__name__)
+
+P = TypeVar("P", bound=Plugin)
+
+
+def _call_safe(func: Callable, args: Tuple, exception_message: str):
```

Comment:
:+1: on the renaming. we should avoid importing too much stuff from localstack into the plugin package, since it will be called from third party tools on `setup.py install`, which would then require all the stuff that localstack.common imports, including six, requests, and dns